src/irr/asset/COBJMeshFileLoader.cpp

```diff
@@ -249,6 +249,8 @@ layout (location = 0) out vec4 OutColor;
 #define map_bump_MASK uint(1u<<10u)
 #define map_normal_MASK uint(1u<<11u)
 
+#ifndef _IRR_FRAG_PUSH_CONSTANTS_DEFINED_
+#define _IRR_FRAG_PUSH_CONSTANTS_DEFINED_
 layout (push_constant) uniform Block {
     vec3 Ka;
```

Comment:
still want all this crap in a struct, furthermore, the alignment of this will be all fucked up, vec3 is aligned like a vec4, I doubt the memory layout is the same in your C++ struct (better put one of the `float`s after each `vec3`)

Reply:
>I doubt the memory layout is the same in your C++ struct

ofc it's not, it's glsl definition, i dont see a problem

Reply:
>I doubt the memory layout is the same in your C++ struct

ofc it's not, it's glsl definition, i dont see a problem

Reply:
see `CMTLPipelineMetadata::SMtl::std140PackedData`

Reply:
> ofc it's not, it's glsl definition, i dont see a problem

It would be a problem if one decided to move the definition from a push constant to an UBO (for example, when rendering multiple materials at once with a single MDI call)

Want to be protected against alignment issues when using the same struct as PC, UBO and SSBO, furthermore even when the struct is a one off or packed in an array.

So ideally structs like this should pass both std140, std430 and be aligned to 16, as well as have their size be aligned to 16 (this protects against differences in layout when you have the struct nested in another as a simple member vs. an array member)